### PR TITLE
First blush at documentation chapter

### PR DESCRIPTION
**Pull request by Hoverbear**
It seemed like when talking about reading APIs it was relevant to talk about how they were generated.

I feel like this is still incomplete, but it covers some basics.

It may not be the best idea to use `iframe`s, so images could be used instead.

English & bad German. :)

### CODE REVIEW

File: presentation/chapters/de-DE/documentation.chapter [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# Dokumentation
+
+---
+
+## `rustdoc`
+
+Rust bietet ein Standarddokumentationswerkzeug namens 'rustdoc`. Es wird häufig durch `cargo doc` verwendet.
+
+Aufgrund dieser Rust-Code ist fast immer in einem gemeinsamen Format dokumentiert.
```
Comment (skade) [inline presentation/chapters/de-DE/documentation.chapter]: "Deshalb ist Rust-Code fast immer in einem gemeinsamen Format dokumentiert"

File: presentation/chapters/de-DE/documentation.chapter [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+`//!` Und `///` Kommentare werden als Markdown gelesen.
+
+```rust
+//! Moduldokumentationskommentare wie der Abschnitt 'Examples' von `std::vec`.
+
+/// Dokumente Funktionen, `struct`s und Eigenschaften.
+/// Hier dokumentiert es eine Funktion.
+fn function_with_documentation() {}
+
+// Zeigt auch in der Dokumentation, aber nicht mit einem Kommentar.
```
Comment (skade) [inline presentation/chapters/de-DE/documentation.chapter]: Ich verstehe diesen Satz nicht.

File: presentation/chapters/de-DE/documentation.chapter [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+`rustdoc` kann Rust-Code und Markdown-Dokumente lesen.
+
+`//!` Und `///` Kommentare werden als Markdown gelesen.
+
+```rust
+//! Moduldokumentationskommentare wie der Abschnitt 'Examples' von `std::vec`.
+
+/// Dokumente Funktionen, `struct`s und Eigenschaften.
+/// Hier dokumentiert es eine Funktion.
```
Comment (skade) [inline presentation/chapters/de-DE/documentation.chapter]: "Hier wird eine Funktion dokumentiert"

**Comment by skade** [comment]
This good, the "reading apis" chapter is also meant as a general chapter to learn how to read out the Rust notation out load.

(like, "what does fn foo<P: AsRef<Path>>(p: P) -> File" mean?)

**Comment by Hoverbear** [comment]
Well, in terms of reading function signatures I figured this would land in the functions/generics chapter. I think this circles back around to chapter ordering. I'm going to try to look at what that should be today.

File: presentation/chapters/de-DE/documentation.chapter (hunk 2 above)
Comment (Hoverbear) [inline presentation/chapters/de-DE/documentation.chapter]: I'm trying to explain that the comment with `//` won't show up in the docs, but the function will.

**Comment by skade** [comment]
Yeah, maybe the example was a bit bad. It was meant as an example chapter of idiomatic, but not simple to read signatures.

(The get/set/Entry-API of HashMap comes to mind)

**Comment by Hoverbear** [comment]
Ah, good idea. I will add some about that.

**Pull request merged**